napari/components/viewer_model.py

```diff
@@ -330,20 +330,11 @@ def _on_layers_change(self, event):
             self.dims.ndim = 2
             self.dims.reset()
         else:
-            extent = self.layers.extent
-            world = extent.world
-            ss = extent.step
-            ndim = world.shape[1]
+            ranges = self.layers._ranges
```

Comment:
We now get the ranges directly from `LayerList._ranges`, so the code below here is now simpler